### DISCUSSION TITLE
Use in domain layer

### DISCUSSION THREAD

**Discussion by hamedsharifi** [PatilShreyas/permission-flow-android#12] [General]
Hi Patil,
Great work. My question is could we use this lib in domain layer of Clean architecture without make depend it to platform?

**Comment by PatilShreyas** [comment]
Yes, it can be used since it doesn't **directly** bring platform components

**Reply by PatilShreyas** [reply]
So the thing is, this library exposes an interface which can be easily plugged into the Domain layer class and can be injected from the constructor. It can be also tested in unit tests by mocking/faking.
